File: InsertionSort/InsertionSort.py
import random
def randomnumber(a):
    for i in range(len(a)):
     a[i]=random.randint(1,100)
def printarray(a):
   for i in range(len(a)):
     print(a[i], end=" ")
def InsertionSort(a):
   for i in range(1,len(a)):
      temp=a[i]
      j=i-1
      while j>=0 and temp<a[j]:
         a[j+1]=a[j]
         j=j-1
      a[j+1]=temp

array=[0]*15
randomnumber(array)
print("Arreglo Original: ")
printarray(array)
InsertionSort(array)
print("\nArray Ordenado:")
printarray(array)
    

File: InsertionSort/InsertionSort.cs

void randomnumber(int[] a)
{
    Random rand = new Random();
    for (int i = 0; i < a.Length; i++)
    {
        a[i] = rand.Next(1, 100);
    }
}

void printarray(int[] a)
{
    for (int i = 0; i < a.Length; i++)
    {
        Console.Write(a[i] + " ");
    }
}

void InsertionSort(int[] a)
{
    for (int i = 1; i < a.Length; i++)
    {
        int temp = a[i];
        int j = i - 1;
        while (j >= 0 && temp < a[j])
        {
            a[j + 1] = a[j];
            j = j - 1;

        }
        a[j + 1] = temp;
    }
}

int[] arr = new int[15];
randomnumber(arr);
Console.WriteLine("Array Original: ");
printarray(arr);
InsertionSort(arr);
Console.WriteLine("\nArray Ordenado: ");
printarray(arr);